modules/renter/contractor/persist.go

```diff
@@ -120,6 +133,26 @@ func (c *Contractor) load() error {
 		}
 	}
 
+	// COMPATv1.1.0
+	//
+	// Wwhen loading old contracts, the corresponding relationships may not be
+	// known. Use the hostdb to make sure the relationship set is fully filled
+	// out.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate backwards (lowest score to highest) so that in the event of
+	// duplicate hosts for a netaddress, the highest score host is the one in
+	// the map.
+	for i := len(allHosts) - 1; i >= 0; i-- {
+		hmap[allHosts[i].NetAddress] = allHosts[i]
+	}
```

Comment:
that's ok, but it's only a weak guarantee. Reminds me of the [HostDBScan thing](https://github.com/NebulousLabs/Sia/blob/master/modules/renter/contractor/uptime.go#L58) where we decided to add a sanity check. But we can't do a sanity check here because we don't have access to the sorting criteria.

Perhaps `AllHosts` should do the filtering internally, so that it is guaranteed to return only one host per NetAddress?